### PR TITLE
mgr/cephadm: Activate existing OSDs

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fixes: https://tracker.ceph.com/issues/49159

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by mgfritch — APPROVED** [review]
seems ok in general, but what about pre-octopus OSDs? ... should we prevent using this in that case and force an adopt?

File: src/pybind/mgr/cephadm/tests/test_cephadm.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def test_apply_osd_save(self, _run_cephadm, cephadm_module: CephadmOrchestrator)
                     '--no-auto', '/dev/sdb', '--yes', '--no-systemd'],
                 env_vars=['CEPH_VOLUME_OSDSPEC_AFFINITY=foo'], error_ok=True, stdin='{"config": "", "keyring": ""}')
             _run_cephadm.assert_called_with(
-                'test', 'osd', 'ceph-volume', ['--', 'lvm', 'list', '--format', 'json'])
+                'test', 'osd', 'ceph-volume', ['--', 'lvm', 'list', '--format', 'json'], image='', no_fsid=False)
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/tests/test_cephadm.py:436]:
```
E           AssertionError: Expected call: _run_cephadm('test', 'osd', 'ceph-volume', ['--', 'lvm', 'list', '--format', 'json'])
E           Actual call: _run_cephadm('test', 'osd', 'ceph-volume', ['--', 'lvm', 'list', '--format', 'json'], image='', no_fsid=False)

**Comment by sebastian-philipp** [comment]
```
>           raise AssertionError(_error_message()) from cause
E           AssertionError: Expected call: _run_cephadm('test', 'osd', 'ceph-volume', ['--', 'lvm', 'list', '--format', 'json'])
E           Actual call: _run_cephadm('test', 'osd', 'ceph-volume', ['--', 'lvm', 'list', '--format', 'json'], image='', no_fsid=False)

/usr/lib/python3.6/unittest/mock.py:814: AssertionError
------------------------------ Captured log call -------------------------------
ERROR    cephadm.services.osd:osd.py:315 Cannot decode JSON: ''
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/cephadm/services/osd.py", line 313, in find_destroyed_osds
    tree = json.loads(out)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
=========================== short test summary info ============================
FAILED cephadm/tests/test_cephadm.py::TestCephadm::test_apply_osd_save_non_collocated

**Comment by sebastian-philipp** [comment]
changelog: 

* PEP-8
* fix make check

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-03-02_10:27:34-rados:cephadm-wip-swagner-testing-2021-03-01-1657-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Pull request merged**